cmd/kubeadm/app/cmd/phases/init/waitcontrolplane.go

```diff
@@ -93,8 +101,11 @@ func runWaitControlPlanePhase(c workflow.RunData) error {
 	fmt.Printf("[wait-control-plane] Waiting for the kubelet to boot up the control plane as static Pods from directory %q. This can take up to %v\n", data.ManifestDir(), timeout)
 
 	if err := waiter.WaitForKubeletAndFunc(waiter.WaitForAPI); err != nil {
+		runtimeSock := strings.Split(data.Cfg().NodeRegistration.CRISocket, "/")
 		ctx := map[string]string{
 			"Error": fmt.Sprintf("%v", err),
+			"Socket": data.Cfg().NodeRegistration.CRISocket,
+			"RuntimeSock": runtimeSock[len(runtimeSock)-1],
```

Comment:
`len(runtimeSock)` can be 0?

Reply:
I cant think of any case in which this could lead to a slice of 0 elements.

You can pass the crio-socket option, but even an empty value (which is not allowed) results in a slice of 1 element.

`nodeRegistration.criSocket: Invalid value: "notAFullPath.sock": path is not absolute: notAFullPath.sock`

Maybe if the config is not properly loaded? But in that case that should be a validation issue, and the program flow should never arrive here.